https://bugs.gentoo.org/show_bug.cgi?id=337025
http://hackage.haskell.org/trac/ghc/ticket/4252
Mon Aug 16 17:24:42 EEST 2010  Ross Paterson <ross@soi.city.ac.uk>
  * fix FP_CHECK_ALIGNMENT for autoconf 2.66 (fixes #4252)
  
  Recent versions of AS_LITERAL_IF don't like *'s.  Fix from
  
  http://blog.gmane.org/gmane.comp.sysutils.autoconf.general/month=20100701
diff -rN -u old-ghc/aclocal.m4 new-ghc/aclocal.m4
--- old-ghc/aclocal.m4	2010-09-13 21:00:36.288338228 +0300
+++ new-ghc/aclocal.m4	2010-09-13 21:00:37.209338228 +0300
@@ -141,7 +141,7 @@
 # A variation of AC_CHECK_SIZEOF for computing the alignment restrictions of a
 # given type. Defines ALIGNMENT_TYPE.
 AC_DEFUN([FP_CHECK_ALIGNMENT],
-[AS_LITERAL_IF([$1], [],
+[AS_LITERAL_IF(m4_translit([[$1]], [*], [p]), [],
                [AC_FATAL([$0: requires literal arguments])])[]dnl
 AC_CHECK_TYPE([$1], [], [], [$3])[]dnl
 m4_pushdef([fp_Cache], [AS_TR_SH([fp_cv_alignment_$1])])[]dnl
